translate.c

```diff
@@ -53,6 +53,20 @@
 #define TQFTPSERV_TMP	"/data/vendor/tmp/tqftpserv"
 #endif
 
+static void read_fw_path_from_sysfs(char *outbuffer, size_t bufsize)
+{
+	size_t pathsize;
+	FILE *f = fopen("/sys/module/firmware_class/parameters/path", "rt");
+	if (!f)
+		return;
+	pathsize = fread(outbuffer, sizeof(char), bufsize, f);
+	fclose(f);
+	if (pathsize <= 0)
```

Comment:
not sure if it can be negative

Reply:
indeed return value type of fread is size_t, so `== 0` it is